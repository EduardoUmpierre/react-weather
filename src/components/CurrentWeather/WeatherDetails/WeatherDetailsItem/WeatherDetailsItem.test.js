import React from 'react'
import { shallow } from 'enzyme'
import WeatherDetailsItems from './WeatherDetailsItem'

describe('<WeatherDetailsItems />', () => {
    it('renders correctly', () => {
        const component = shallow(
            <WeatherDetailsItems>10ºC</WeatherDetailsItems>
        )
        expect(component).toMatchSnapshot()
    })

    it('has bold text', () => {
        const component = shallow(
            <WeatherDetailsItems highlighted>10ºC</WeatherDetailsItems>
        )
        expect(component.prop('style')).toHaveProperty('fontWeight', 700)
    })

    it('does not have bold text', () => {
        const component = shallow(
            <WeatherDetailsItems>10ºC</WeatherDetailsItems>
        )
        expect(component.prop('style')).not.toHaveProperty('fontWeight', 700)
    })
})
