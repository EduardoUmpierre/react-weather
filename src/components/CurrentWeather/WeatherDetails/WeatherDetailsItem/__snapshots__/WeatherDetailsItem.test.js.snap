// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherDetailsItems /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <weatherDetailsItem>
    10ºC
  </weatherDetailsItem>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="weatherDetailsItemTitle"
        />,
        "10ºC",
      ],
      "className": "weatherDetailsItem",
      "style": Object {
        "fontWeight": null,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "weatherDetailsItemTitle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "10ºC",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="weatherDetailsItemTitle"
          />,
          "10ºC",
        ],
        "className": "weatherDetailsItem",
        "style": Object {
          "fontWeight": null,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "weatherDetailsItemTitle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "10ºC",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
